Formula/libtorch.rb

```diff
@@ -58,9 +58,11 @@ def install
       system "cmake", "..", *std_cmake_args, *args
 
       # Avoid references to Homebrew shims
-      inreplace "caffe2/core/macros.h",
-                "{\"CXX_COMPILER\", \"#{HOMEBREW_SHIMS_PATH}/mac/super/clang++\"},",
-                "{\"CXX_COMPILER\", \"/usr/bin/clang++\"},"
+      on_macos do
+        inreplace "caffe2/core/macros.h",
+                  "{\"CXX_COMPILER\", \"#{HOMEBREW_SHIMS_PATH}/mac/super/clang++\"},",
+                  "{\"CXX_COMPILER\", \"/usr/bin/clang++\"},"
+      end
```

Comment:
```suggestion
      inreplace "caffe2/core/macros.h", %r{#{HOMEBREW_SHIMS_PATH}/[^/]/super}, ""
```

Reply:
Or, if you prefer something stricter:
```suggestion
      inreplace "caffe2/core/macros.h", %r{#{HOMEBREW_SHIMS_PATH}/[^/]/super/#{ENV.cxx}}, ENV.cxx
```

Reply:
The 2nd may be safer, but will need to be `Regexp.escape`'d due to `++` being special symbols.